docs/proposals/hpa/federated-hpa.md

```diff
@@ -0,0 +1,286 @@
+
+Every time the `spec.replicas` of Workload in the control plane is modified, the replicas distribution in the `Work` will be re-calculated by Karmada scheduler. The replicas distribution calculated by Karmada scheduler most probably don't match the actual distribution in the member clusters. The mismatch need to be solved to get rid of the risk of incident mentioned above.
+
+### How to deal with the situation that `spec.replicas` is `1`
+The workload can be spread into multiple member clusters when the `spec.replicas` in the control plane is greater than `1`. The disaster of one member cluster and control plane can be tolerated because the workload can be scaled in other member clusters. But if the `spec.replicas` is `1`, the workload and HPA resource would only be spread into one member cluster. If the member cluster and control plane are out of service in the same time, the workload can't be scaled.
+
+### How to integrate with and migrate from existing HPA resources
+For some scenarios, people may want a friendly mechanism to control what HPA resources can be controled by `FederatedHPAController`
+1. There are already many HPA resources in the control plane managed by `PropagationPolicy` and `OverridePolicy` before the Karmada support FederatedHPA natively. For some risk concerns, the administrator of the platform wants to migrate these HPA resources to be managed by the FederatedHPAController step by step. 
+1. There are already many HPA resources in the control plane managed by `PropagationPolicy` and `OverridePolicy` before the Karmada support FederatedHPA natively. But in the same Karmada control plane, some users want to use the native FederatedHPA but others want to remain the old ways. The FederatedHPA should not conflict with the HPA resources managed by the old ways.
+
+To meet the requirements of above scenarios mentioned, we can introduce a field `enabledByDefault` to FederatedHPA configuration and a label `hpa.karmada.io/enabled=true/false` to HPA resources
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: federated-hpa
+  namespace: karmada-system
+data:
+# ...
+  enabledByDefault: true/false
+```
+
+If the global `enabledByDefault` is set to `true`, `FederatedHPAController` will take control of all the HPA resources except which is labeled `hpa.karmada.io/enabled=false`. 
```

Comment:
I suggest just introducing the `federatedhpa.karmada.io/enabled` label for alpha.
- federatedhpa.karmada.io/enabled=true means the HPA is FederatedHPA and it'll be processed by FederatedHPAController.
- federatedhpa.karmada.io/enabled=false means the HPA is `normal` HPA and it'll be propagated by PP/CPP.

Reply:
@RainbowMango  Do you mean remove the `enabledByDefault` global configuration? If so, what if HPA resources without label `federatedhpa.karmada.io/enabled` ? Is it regarded as a normal HPA or federated HPA resource? For scenario which only uses FederatedHPA, it would be too tedious to label every HPA resources instead of using global `enabledByDefault`.

Reply:
> Do you mean remove the enabledByDefault global configuration?

Yes. 

>  For scenario which only uses FederatedHPA, it would be too tedious to label every HPA resources instead of using global enabledByDefault.

Yes, I agree with you. Can we tolerate it at the `alpha` stage?

I'm hesitant to introduce a configuration by `Secret`, as the configuration essentially is a kind of API. And the API group [config.karmada.io](https://github.com/karmada-io/karmada/tree/master/pkg/apis/config) is the appropriate place.

Yeah, I also agree to start to design the config API, but I'm afraid it'll slow down the progress.

Reply:
@RainbowMango It makes sense, I agree with you.

Reply:
By the way, are you going to present at the community meeting today? I guess we can talk about how to dev plan now.